From 658d387562fb63459e83f3c1b8b31b51a368ccfe Mon Sep 17 00:00:00 2001
From: Siu Kwan Lam <1929845+sklam@users.noreply.github.com>
Date: Mon, 9 Jun 2025 15:41:24 -0500
Subject: [PATCH 2/2] Fix list-comprehension for python 3.13.4

---
 numba/core/bytecode.py | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/numba/core/bytecode.py b/numba/core/bytecode.py
index 3df9c06bd..c312c8fe8 100644
--- a/numba/core/bytecode.py
+++ b/numba/core/bytecode.py
@@ -2,6 +2,7 @@ from collections import namedtuple, OrderedDict
 import dis
 import inspect
 import itertools
+import sys
 
 from types import CodeType, ModuleType
 
@@ -538,8 +539,12 @@ class ByteCodePy312(ByteCodePy311):
                 # If we see a GET_ITER here, check if the next thing is a
                 # FOR_ITER.
                 if next_inst.opname == "GET_ITER":
-                    # Add the inst to potentially be replaced to NOP
-                    current_nop_fixes.add(next_inst)
+                    # In Python 3.13.4, this becomes the only GET_ITER,
+                    # so don't turn it into a NOP.
+                    # Python 3.13.5 reverted the change.
+                    if sys.version_info[:3] != (3, 13, 4):
+                        # Add the inst to potentially be replaced to NOP.
+                        current_nop_fixes.add(next_inst)
                     # Loop up next instruction.
                     next_inst = self.table[self.ordered_offsets[index + 3]]