--- numba/_helperlib.c	2025-01-30 13:10:43.125525098 +0000
+++ "numba/_helperlib copy.c"	2025-01-30 14:58:49.481280234 +0000
@@ -19,7 +19,7 @@
 #else
     #include <stdint.h>
     #define _complex_float_t complex float
-    #define _complex_float_ctor(r, i) (r + I * i)
+    #define _complex_float_ctor(r, i) ((complex float){(r), (i)})
     #define _complex_double_t complex double
 #endif
 #define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION
